### ISSUE TITLE
Cart Items changes track

### ISSUE THREAD

**Issue by codewp** [woocommerce/woocommerce-gutenberg-products-block#5414]
The cart block can not track changes in cart items, for example, quantity changes of an item.

In the below screenshot, the user changes the first item quantity for "3" to "4" and the second item quantity will change in the backend by an auto-add plugin, which automatically adds items to the cart like in the BOGO deals.

![5](https://user-images.githubusercontent.com/10684451/146679523-7d1dff62-bca0-434a-b5c6-96a752141e5f.png)

On the other hand, it can track cart items' overall change, like an item removed from the cart in the backend.
Please check the below gif video for more information. It is a Buy 2 Get 1 Free deal.

![Record_2021_12_19_18_32_42_774](https://user-images.githubusercontent.com/10684451/146679908-669480d0-cf3a-4e79-a8a6-522d665a4866.gif)

As you can see in the above video the cart block can track item add or remove changes but it can not track changes in items quantity.

**Comment by opr** [comment]
Hi there, I'm unable to reproduce this, but I'll ask some other members of the team to try as well.

Using this code to add 1 to the quantity of every other item in the cart when the quantity of any item is updated.

I added this to my theme's `functions.php` file

```php
/**
 * Updates all other items in cart to have +1 quantity
 */
add_action('woocommerce_after_cart_item_quantity_update', function( $cart_item_key, $quantity, $old_quantity, $cart ) {
	$keys = array_filter( array_keys( $cart->cart_contents ), function( $key ) use ( $cart_item_key ) { return $key !== $cart_item_key; } );
	foreach( $keys as $key ) {
		$cart->cart_contents[ $key ]['quantity'] += 1;
	}
}, 10, 4);
```

Here's a video of the cart updating correctly.
https://www.loom.com/share/8e34c728ab0a41e2bfbea9b145aebbfa

In the meantime could you share your [WC status report](https://woocommerce.com/document/understanding-the-woocommerce-system-status-report/) with us?

**Comment by mikejolley** [comment]
I think support for this was added in 6.6.0?

**Comment by codewp** [comment]
@opr and @mikejolley 

Thank you for your feedback.

Yes, it seems that it is fixed in 6.6.0
I tested it on 6.5 before the 6.6 release.

Thanks
